yt/geometry/particle_oct_container.pyx

```diff
@@ -1752,11 +1751,14 @@ cdef class ParticleBitmapSelector:
                                np.uint64_t ind1[3]) except -1:
         cdef np.uint64_t imi, fmi
         cdef np.uint64_t mi
-        cdef np.uint64_t indexgap = 1 << (self.bitmap.index_order1 - nlevel)
-        imi = encode_morton_64bit(ind1[0], ind1[1], ind1[2])
-        fmi = encode_morton_64bit(
-            ind1[0]+indexgap-1, ind1[1]+indexgap-1, ind1[2]+indexgap-1)
-        for mi in range(imi, fmi+1):
+        cdef np.uint64_t start_ind[3], end_ind[3]
+        cdef np.uint64_t shift_by = (self.bitmap.index_order1 - nlevel)
+        for i in range(3):
+            start_ind[i] = ind1[i] << shift_by
+            end_ind[i] = start_ind[i] + (1 << shift_by) - 1
+        imi = encode_morton_64bit(start_ind[0], start_ind[1], start_ind[2])
+        fmi = encode_morton_64bit(end_ind[0], end_ind[1], end_ind[2])
+        for mi in range(imi, fmi):
```

Comment:
If the morton indices are calculated in this way, I think the loop should be inclusive of `fmi`, which is the last morton index in the division identified by `ind1`. It can be left as non-inclusive if the version from my comment w/ the suggestion is used because then `fmi` will be the first index in the next morton ordered division.

Reply:
Thank you for the suggestions!  I went ahead and committed all three of them.